CRM/Custom/Form/Option.php

```diff
@@ -473,16 +447,18 @@ public function postProcess() {
       }
 
       if (!empty($params['default_value'])) {
-        $customField->default_value = $customOption->value;
-        CRM_Core_BAO_CustomField::createField($customField, 'modify');
+        $newDefaultValue = $customOption->value;
       }
       elseif ($customField->find(TRUE) && $customField->default_value == $customOption->value) {
         // this is the case where this option is the current default value and we have been reset
-        $customField->default_value = 'null';
-        CRM_Core_BAO_CustomField::createField($customField, 'modify');
+        $newDefaultValue = '';
       }
     }
 
+    if (isset($newDefaultValue)) {
+      CRM_Core_BAO_CustomField::create(['id' => $this->_fid, 'default_value' => $newDefaultValue]);
+    }
```

Comment:
This is the heart of the fix: calling `create` instead of `createField`. The latter does not actually update the row in `civicrm_custom_field` so the 'default_value' was not getting saved. 

@larssandergreen